setuptools/tests/test_build_meta.py

```diff
@@ -184,24 +185,10 @@ def test_build_sdist_version_change(build_backend):
 
 
 def test_build_meta_setup_py_exists(tmpdir_cwd):
-    # if build_sdist(".") is called, ensure setup.py is included
+    # if build_sdist is called not from a script other than setup.py, ensure setup.py is included
     import tarfile
-    from setuptools.build_meta import build_sdist
     build_files(defns[0])
-    targz_path = build_sdist(".")
-    tar = tarfile.open(targz_path)
+    targz_path = build_sdist("temp")
+    tar = tarfile.open(os.path.join("temp", targz_path))
     assert any('setup.py' in name for name in tar.getnames())
 
-
-@pytest.mark.xfail
```

Comment:
Why the removal? is this not the desired behavior?

Reply:
That test was my misunderstanding of the [sdist_directory](https://www.python.org/dev/peps/pep-0517/#build-sdist) argument. I thought it meant source directory of the sdist (source-tree) and thus wrote this test that expected `build_sdist("/package/src")` and it should build an sdist from that directory.

On reading PEP517, I understood it means 'destination_directory` for placing the sdist and thus this test stood invalid. I guess the PEP implies that the build_meta commands will always be executed from the source_tree as the cwd() (containing setup.py). 

All other tests are valid though, that test it from the cwd() but with a different script name.